DOC,REL: Update 1.11.0 notes.

charris opened the pull request:
The indexing deprecation warnings to exception changes originally
scheduled for 1.11.0 have been reverted in order to allow downstream
projects more time to make releases fixing the problem.

rgommers commented on doc/sphinxext:
```diff
@@ -1 +1 @@
-Subproject commit ef988a4a4658c991f4445f6241ab02d74710c6e3
+Subproject commit 84cc897d266e0afc28fc5296edf01afb08005472
```
Accidental change (you forgot ``git submodule update`` and did ``git commit -a``).

charris committed:
DOC,REL: Update 1.11.0 notes.

The indexing deprecation warnings to exception changes originally
scheduled for 1.11.0 have been reverted in order to allow downstream
projects more time to make releases fixing the problem.

charris commented:
Fixed.

rgommers commented:
LGTM, merged. Thanks Chuck.
